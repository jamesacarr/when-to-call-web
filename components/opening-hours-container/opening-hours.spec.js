import React from 'react';
import { shallow } from 'enzyme';
import OpeningHours from './opening-hours';

describe('OpeningHours', () => {
  const offset = new Date().getTimezoneOffset() * -1;
  const classes = {
    row: 'row',
    cell: 'cell',
  };

  it('renders nothing when no open', () => {
    const data = { close: { day: 0, time: '1700' } };
    const wrapper = shallow(<OpeningHours classes={classes} data={data} offset={offset} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders nothing when no close', () => {
    const data = { open: { day: 0, time: '0900' } };
    const wrapper = shallow(<OpeningHours classes={classes} data={data} offset={offset} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders nothing when no intersection', () => {
    const data = {
      open: { day: 0, time: '1800' },
      close: { day: 0, time: '2000' },
    };
    const wrapper = shallow(<OpeningHours classes={classes} data={data} offset={offset} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const data = {
      open: { day: 0, time: '1000' },
      close: { day: 0, time: '1400' },
    };
    const wrapper = shallow(<OpeningHours classes={classes} data={data} offset={offset} />);

    expect(wrapper).toMatchSnapshot();
  });
});
