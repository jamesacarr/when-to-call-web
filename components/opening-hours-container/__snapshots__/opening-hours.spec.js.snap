// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningHours renders correctly 1`] = `
<WithStyles(TableRow)
  className="row"
>
  <WithStyles(TableCell)
    className="cell"
    padding="none"
  >
    Sunday
  </WithStyles(TableCell)>
  <WithStyles(TableCell)
    className="cell"
    padding="none"
  >
    10:00AM
     - 
    02:00PM
  </WithStyles(TableCell)>
</WithStyles(TableRow)>
`;

exports[`OpeningHours renders nothing when no close 1`] = `""`;

exports[`OpeningHours renders nothing when no intersection 1`] = `""`;

exports[`OpeningHours renders nothing when no open 1`] = `""`;
